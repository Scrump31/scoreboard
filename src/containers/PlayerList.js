import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import { removePlayer, updatePlayerScore } from '../actions';

import Player from '../components/Player';

const PlayerList = ({ players, deletePlayer, updateScore }) => (
  <div>
    <ReactCSSTransitionGroup
      transitionName={'fade'}
      transitionEnterTimeout={500}
      transitionLeaveTimeout={500}
    >
      {players.map(player => (
        <Player
          name={player.name}
          score={player.score}
          key={player.id}
          id={player.id}
          deletePlayer={deletePlayer}
          updateScore={updateScore}
        />
      ))}
    </ReactCSSTransitionGroup>
  </div>
);
const mapDispatchToProps = dispatch => ({
  deletePlayer: id => dispatch(removePlayer(id)),
  updateScore: (newScore, id) => dispatch(updatePlayerScore(newScore, id)),
});

const mapStateToProps = ({ players }) => ({ players });

PlayerList.propTypes = {
  players: PropTypes.arrayOf(PropTypes.object).isRequired,
  deletePlayer: PropTypes.func.isRequired,
  updateScore: PropTypes.func.isRequired,
};
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PlayerList);
