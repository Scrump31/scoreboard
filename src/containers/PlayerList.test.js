import React from 'react';
import { shallow, mount } from 'enzyme';
import { PlayerList } from './PlayerList';
import PLAYERS from '../mock/mock';

const PLAYER_ROW = '.row .player-name.col-xs-9';

describe('<PlayerList />', () => {
  const mockDelPlayer = jest.fn();
  const mockUpdateScore = jest.fn();

  const mockPlayers = PLAYERS;
  it('renders without crashing', () => {
    const component = shallow(
      <PlayerList
        players={mockPlayers}
        updateScore={mockUpdateScore}
        deletePlayer={mockDelPlayer}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('creates 3 Players given an array of 3 player objects ', () => {
    const component = mount(
      <PlayerList
        players={mockPlayers}
        updateScore={mockUpdateScore}
        deletePlayer={mockDelPlayer}
      />,
    );
    expect(component.find(PLAYER_ROW).length).toEqual(3);
    component.unmount();
  });
});
