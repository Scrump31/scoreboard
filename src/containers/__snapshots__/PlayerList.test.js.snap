// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerList /> renders without crashing 1`] = `
<TransitionGroup
  childFactory={[Function]}
  component="div"
>
  <CSSTransition
    classNames="fade"
    key="1"
    timeout={500}
  >
    <Player
      deletePlayer={[Function]}
      id="1"
      key="1"
      name="Player One"
      score={31}
      updateScore={[Function]}
    />
  </CSSTransition>
  <CSSTransition
    classNames="fade"
    key="2"
    timeout={500}
  >
    <Player
      deletePlayer={[Function]}
      id="2"
      key="2"
      name="Player Two"
      score={35}
      updateScore={[Function]}
    />
  </CSSTransition>
  <CSSTransition
    classNames="fade"
    key="3"
    timeout={500}
  >
    <Player
      deletePlayer={[Function]}
      id="3"
      key="3"
      name="Player Three"
      score={42}
      updateScore={[Function]}
    />
  </CSSTransition>
</TransitionGroup>
`;
